client/src/components/game/lab4/pages/CodeChangeAccessible.js

```diff
@@ -0,0 +1,277 @@
+            this.state = {
+                textValue: window.location.state.hint,
+                snackBarOpen: false,
+                message: "Please type code before updating code!"
+            };
+            window.location.state = {
+                hint: window.location.state.hint
+            };
+        } else {
+            window.location.state = {
+                hint: null
+            };
+        }
+    }
+
+    componentDidMount() {
+        const { actions } = this.props;
+        actions.updateState(GAME_PLAYING);
+        Prism.highlightAll();
+        if (window.location.state.hint !== undefined) {
+            const el0 = document.getElementById("first");
+            el0.value = window.location.state.hint;
+            CodeChangeAccessible.doEvent(el0, "input");
+        }
+    }
+
+    handleChange(event) {
+        this.setState({ textValue: event.target.value }, () => {
+            console.log("handled change value: " + this.state.textValue);
+            Prism.highlightAll();
+        });
+    }
+
+    handleClose(event, reason) {
+        if (reason === "clickaway") {
+            return;
+        }
+
+        this.setState({ snackBarOpen: false }, () => {
+            console.log("SnackBar Closed");
+        });
+    }
+
+    handleSubmit(event) {
+        event.preventDefault();
+        console.log("hint updated as: " + this.state.textValue);
+        if (this.state.textValue === "") {
+            this.setState({
+                message: "Please type code before updating code!",
+                snackBarOpen: true
+            });
+        } else if (parseInt(this.state.textValue) !== 0) {
+            this.setState({ message: "Please enter value 0", snackBarOpen: true });
+        } else if (!/^\d+$/.test(this.state.textValue)) {
+            this.setState({
+                message: "Please enter numeric value",
+                snackBarOpen: true
+            });
+        } else {
+            window.location.state = {
+                hint: this.state.textValue
+            };
+            navigate("/Lab4/Game/FormHintAccessible");
+        }
+        Prism.highlightAll();
+    }
+
+    static doEvent(obj, event) {
+        const eventInit = new Event(event, { target: obj, bubbles: true });
+        return obj ? obj.dispatchEvent(eventInit) : false;
+    }
+
+    render() {
+        const paperStyle = {
+            marginLeft: "10px",
+            marginRight: "10px",
+            marginTop: "20px"
+        };
+        return (
+            <div>
+                {/* <CodeUpdateHeader
+                    heading={"Make Code Changes"}
+                    justifyAlignment={"space-between"}
+                    helpMessage={"Set the tabindex to 0 in the code editor."}
+                /> */}
+                <div className="app__instructions__small2">
+                    <h2>Repair</h2><br/>
+                    The intent of this code repair is to ensure that, wherever possible,
```

Comment:
Maybe consider shortening this - This is a lot to read and I naturally just skimmed it